packages/server/graphql/mutations/helpers/notifications/notifyMattermost.ts

```diff
@@ -0,0 +1,298 @@
+import {SlackNotificationEventEnum as EventEnum} from '../../../../database/types/SlackNotification'
+import makeAppURL from 'parabol-client/utils/makeAppURL'
+import appOrigin from '../../../../appOrigin'
+import MattermostServerManager from '../../../../utils/MattermostServerManager'
+import segmentIo from '../../../../utils/segmentIo'
+import sendToSentry from '../../../../utils/sendToSentry'
+import {DataLoaderWorker} from '../../../graphql'
+import {
+  makeFieldsAttachment,
+  makeHackedFieldButtonValue,
+  makeHackedButtonPairFields
+} from './makeMattermostAttachments'
+import getSummaryText from './getSummaryText'
+import MeetingRetrospective from '../../../../database/types/MeetingRetrospective'
+import MeetingAction from '../../../../database/types/MeetingAction'
+import MeetingPoker from '../../../../database/types/MeetingPoker'
+import findStageById from 'parabol-client/utils/meetings/findStageById'
+import {phaseLabelLookup} from 'parabol-client/utils/meetings/lookups'
+import formatWeekday from 'parabol-client/utils/date/formatWeekday'
+import formatTime from 'parabol-client/utils/date/formatTime'
+import {toEpochSeconds} from '../../../../utils/epochTime'
+
+const getWebhookForUserIdTeamId = async (
+  userId: string,
+  teamId: string,
+  dataLoader: DataLoaderWorker
+) => {
+  const mattermost = await dataLoader.get('mattermostBestAuthByUserIdTeamId').load({userId, teamId})
+  return mattermost?.webhookUrl
+}
+
+const notifyMattermost = async (
+  event: EventEnum,
+  webhookUrl: string,
+  userId: string,
+  teamId: string,
+  textOrAttachmentsArray: string | unknown[],
+  notificationText?: string
+) => {
+  const manager = new MattermostServerManager(webhookUrl)
+  const result = await manager
+    .postMessage(textOrAttachmentsArray, notificationText)
+    .catch((err) => new Error(`Mattermost error: ${err.message}`))
```

Comment:
the only error that throws on this is:

```ts
    if (!prop) throw new Error('Invalid mattermost message')
```

which isn't a mattermost error, it's an error we make due to a bad `textOrAttachmentsArray`.

let's have `postMessage` always return an error & never throw & then we can just catch that in the line below

Reply:
Fixed!